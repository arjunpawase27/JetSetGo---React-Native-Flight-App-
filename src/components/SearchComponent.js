import React, { useState } from 'react';
import { View, TextInput, Button, StyleSheet, Image, TouchableOpacity, Keyboard } from 'react-native';

const SearchComponent = ({ onSearch }) => {
  const [searchQuery, setSearchQuery] = useState('');

  
const handleClear = ()=>{
    setSearchQuery("")
    onSearch("");
    Keyboard.dismiss();
}
  const handleChangeText = (text) => {
  setSearchQuery(text)
    onSearch(text);
  };


  return (
    <View style={styles.searchContainer}>
      <TextInput
        style={styles.searchInput}
        onChangeText={handleChangeText} 
        value={searchQuery}
        placeholder="Search by airline"
        placeholderTextColor="#777"
      />
       {!searchQuery?
       <Image
        source={require('../assets/images/search.jpg')} 
        style={styles.searchIcon}
      />:<TouchableOpacity onPress={handleClear} testID="cancel-button">
        <Image
        source={require('../assets/images/cancel.jpg')} 
        style={styles.searchIcon}
      />
        </TouchableOpacity>}
    </View>
  );
};

const styles = StyleSheet.create({
  searchContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 20,
    borderColor: '#ced4da',
    borderWidth: 1,
    paddingHorizontal: 10,
    borderRadius: 20,
  },
  searchInput: {
    marginRight: 10,
    flex: 1,
    height: 40,
   
    color: '#495057',
  },
});

export default SearchComponent;
