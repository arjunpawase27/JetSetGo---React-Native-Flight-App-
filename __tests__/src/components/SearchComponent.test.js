import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import SearchComponent from '../../../src/components/SearchComponent';
describe('SearchComponent', () => {
  test('renders correctly', () => {
    const { toJSON } = render(<SearchComponent onSearch={() => {}} />);
    expect(toJSON()).toMatchSnapshot();
  });

  test('updates search query on text input change', () => {
    const onSearchMock = jest.fn();
    const { getByPlaceholderText } = render(<SearchComponent onSearch={onSearchMock} />);

    const input = getByPlaceholderText('Search by airline');
    fireEvent.changeText(input, 'Air India');
    
    expect(input.props.value).toBe('Air India');
    expect(onSearchMock).toHaveBeenCalledWith('Air India');
  });

  test('clears search query and calls onSearch when cancel button is pressed', () => {
    const onSearchMock = jest.fn();
    const { getByTestId, getByPlaceholderText } = render(<SearchComponent onSearch={onSearchMock} />);

    const input = getByPlaceholderText('Search by airline');
    fireEvent.changeText(input, 'Air India');

    const cancelButton = getByTestId('cancel-button');
    fireEvent.press(cancelButton);

    expect(input.props.value).toBe('');
    expect(onSearchMock).toHaveBeenCalledWith('');
  });
  
});
