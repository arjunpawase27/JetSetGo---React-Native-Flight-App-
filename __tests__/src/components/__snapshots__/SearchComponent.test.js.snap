// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchComponent renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "borderColor": "#ced4da",
      "borderRadius": 20,
      "borderWidth": 1,
      "flexDirection": "row",
      "marginBottom": 20,
      "paddingHorizontal": 10,
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder="Search by airline"
    placeholderTextColor="#777"
    style={
      {
        "color": "#495057",
        "flex": 1,
        "height": 40,
        "marginRight": 10,
      }
    }
    value=""
  />
  <Image
    source={
      {
        "testUri": "../../../src/assets/images/search.jpg",
      }
    }
  />
</View>
`;
